### ISSUE TITLE
`limit` should be done after the filtering, not before

### ISSUE THREAD

**Issue by DavidStrada** [transformstudios/statamic-events-legacy#40]
Using the following data on `events` collection 

```
title: 'Alexander''s Broadbent Selections Wine Dinner'
multi_day: false
start_date: '2020-10-15'
all_day: false
start_time: '18:30'
end_time: '20:30'
recurrence: false
id: 1804ff56-98bb-4a1b-8bda-1ff88db93212
event_categories:
  - dining
  - featured


-----
title: 'Women''s Golf Day'
multi_day: false
start_date: '2020-09-01'
all_day: false
start_time: '15:00'
end_time: '19:00'
recurrence: false
id: cdca2ca1-d54e-4520-ad97-59389c691419
event_categories:
  - golf
  - featured

```
`event_categories = 'featured'`
`{{ events:upcoming collection="events" limit="6" :taxonomy="event_categories" }}` using the tag like this.

`Women''s Golf Day` never gets displayed, it seems like it's ordering and filtering by `alphabetical` `title` or something, if you change the `title` to `A Women''s Golf Day` it works.

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_2zs5r)

**Comment by DavidStrada** [comment]
Hey there, wanted to check if you guys have an ETA on this bug, when it's going to be address or if it's going to be address at all ? thanks 👍

**Comment by adnankussair** [comment]
Hey David. We plan to look at this on Wednesday of this week. Thanks!

-- 
Adnan Kussair (AK)

Transform Studios | Strategy. Design. Code.

--- original message ---
On August 31, 2020, 8:53 AM PDT notifications@github.com wrote:



Hey there, wanted to check if you guys have an ETA on this bug, when it's going to be address or if it's going to be address at all ? thanks 👍

—

You are receiving this because you are subscribed to this thread.

Reply to this email directly, view it on GitHub, or unsubscribe.
--- end of original message ---

**Comment by adnankussair** [comment]
@DavidStrada Just a heads up that we'll be taking a crack at getting this reported bug resolved for you by tomorrow morning. Today got the best of us. :)

**Comment by edalzell** [comment]
@DavidStrada that's not the correct syntax. You filter the same way you would with the collection tag, `taxonomy:event_categories="featured"`

**Comment by edalzell** [comment]
@DavidStrada which version of Events? The latest is 1.5.1.

**Comment by DavidStrada** [comment]
`version: "1.4"`, I'm testing on the latest to see if the error persists.

**Comment by DavidStrada** [comment]
I've tested this on the latest and I'm still having this issue, I've created a video maybe this will give you a better understanding of what the issue is, https://www.dropbox.com/s/wbbcsha36ecuk3d/events-calendar.mov?dl=0

**Comment by edalzell** [comment]
> @DavidStrada that's not the correct syntax. You filter the same way you would with the collection tag, `taxonomy:event_categories="featured"`

@DavidStrada let's see your template code please.

**Comment by DavidStrada** [comment]
@edalzell here you go : https://gist.github.com/DavidStrada/6c72f8544e00a965a12a5b7658477454

**Comment by DavidStrada** [comment]
btw: tested your syntax `{{ events:upcoming collection="events" limit="6" :taxonomy:event_categories="featured" }}` didn't work either, same result (video).

**Comment by edalzell** [comment]
> btw: tested your syntax `{{ events:upcoming collection="events" limit="6" :taxonomy:event_categories="featured" }}` didn't work either, same result (video).

You've got a colon in there, should be `taxonomy:event_categories="featured"` NOT `:taxonomy:event_categories="featured"`

**Comment by DavidStrada** [comment]
removed the colon and the problem still persist, `if` I add an extra `A` it pulls data otherwise it doesn't so not really sure what's causing that.
`{{ events:upcoming collection="events" limit="6" taxonomy:event_categories="featured" }}`

**Comment by edalzell** [comment]
If you make the limit bigger does it work? like `50` or something?

**Comment by DavidStrada** [comment]
it does work, however besides the point of `50`, if the event is on `September` and others on `October` it should pull e.g.
1 from Sept. (FRI sep 25th)
5 from Oct. (...)

shouldn't it ?

**Comment by DavidStrada** [comment]
btw: `{{ events:upcoming collection="events" limit="50" :taxonomy="event_categories" }}` this works as well if I do the `50` so syntax wise I guess there isn't an issue here per say.

**Comment by edalzell** [comment]
Ya I still think there might be a bug on the limit, it's being done in the wrong spot so it's limiting the wrong things. I'll have to do some more testing, but glad you got it working in the short term. Sorry about that!

**Comment by DavidStrada** [comment]
> Ya I still think there might be a bug on the limit, it's being done in the wrong spot so it's limiting the wrong things. I'll have to do some more testing, but glad you got it working in the short term. Sorry about that!

Yeah, I went with the option of adding an extra `A` so it solves the problem for now, no worries, it's a tricky bug so thanks for having a look at it, and let me know if I can be more helpful 👍

**Comment by edalzell** [comment]
True issue here is that we are `limit`ing **all** the dates, NOT the ones that are in the future, on the `upcoming` tag.

Need to update to filter first, then limit.

@adnankussair we should schedule this into the rotation

**Comment by DavidStrada** [comment]
Hey Guys, wondering if you made any progress on this,

Thanks

**Comment by adnankussair** [comment]
@DavidStrada We have this scheduled to get worked on this week. We'll let you know the moment its resolved. Thanks for your patience!

**Comment by edalzell** [comment]
I can't reproduce this @DavidStrada, are you able to send me your site privately so I can take a look?

Either via DM on Discord or erin@transformstudios.com.

**Comment by DavidStrada** [comment]
Hey Guys, just wondering if you made any progress on this, or if you need further information.
Thanks

**Comment by edalzell** [comment]
I'll be taking a look at this tomorrow. I got your site.

**Comment by edalzell** [comment]
Fixed in 1.5.2.
